### PR TITLE
iio: Check buffer existence before use it

### PR DESCRIPTION
**Pull request by yongli3**
There are several iio device drivers do not support buffer reading,
if enable buffer reading(buffer_size > -1) in fbp files for these devices,
Reading will fail, add buffer checking logic to print warning message and quit the fbp

Signed-off-by: Yong Li <yong.b.li@intel.com>

### CODE REVIEW

**Comment by yongli3** [comment]
@edersondisouza @barbieri @neofang please help to review, this PR is used for #2044

File: src/lib/io/sol-iio.c [hunk 1]
```diff
@@ -639,6 +640,17 @@ sol_iio_open(int device_id, const struct sol_iio_config *config)
...
+    } else {
+        buffer_existence = false;
+        if (config->buffer_size > -1) {
+            SOL_WRN("Buffer is enabled but device%d does not support Buffer.",
+                device->device_id);
+            goto error;
+        }
+    }
+
     if (config->buffer_size > -1) {
```
Comment (glima) [inline src/lib/io/sol-iio.c]: You could have the goto above here instead and have less code.

Comment (yongli3) [inline src/lib/io/sol-iio.c]: @glima I changed the if logic, please check it again

**Comment by glima** [comment]
OK, go for it

**Comment by yongli3** [comment]
@glima do you mean we can merge it? If so please help to merge.

**Comment by edersondisouza** [comment]
+1

**Comment by edersondisouza** [comment]
Merged, tks!

**Pull request closed without merging**